lib/dialects/abstract/connection-manager.js

```diff
@@ -184,11 +184,28 @@ ConnectionManager.prototype.getConnection = function(options) {
   var self = this;
   options = options || {};
 
-  return new Promise(function (resolve, reject) {
-    self.pool.acquire(function(err, connection) {
-      if (err) return reject(err);
-      resolve(connection);
-    }, options.priority, options.type, options.useMaster);
+  return Promise.resolve().then(function () {
+    if (self.sequelize.options.databaseVersion === 0) {
+      return self.$connect(self.config).then(function (connection) {
+        var _options = {};
+        _options.transaction = { connection: connection }; // Cheat .query to use our private connection
+        _options.logging = function () {};
+        _options.logging.__testLoggingFn = true;
+
+        return self.sequelize.databaseVersion(_options).then(function (version) {
+          self.sequelize.options.databaseVersion = version;
+
+          self.$disconnect(connection);
+        });
+      });
+    }
+  }).then(function () {
```

Comment:
A Promise should always [return something] (https://github.com/spion/thenlint) in the middle of a promise chain so I recommend to rewrite this to:
```js
var promise;
if (self.sequelize.options.databaseVersion === 0) {
  promise = self.$connect(self.config)...
} else {
  promise = Promise.resolve();
}
return promise.then(function () { ... });
```
That's also a bit less indentation (and you use one promise less). Some might prefere the other way to write this because of the style though. In that case it would be good to add a `return;` behind the if.